tests/app.test.ts

```diff
@@ -1,8 +1,99 @@
 import request from "supertest";
-import  app from "../source/app";
+import router from '../routes/routes';
+import { Soldier } from "../repository/repo";
+const app = router; 
+beforeEach(async ()=>{ 
+    await Soldier.deleteMany();
+});
 
 describe('Is Soldier Alive', () => {
     it('Should Send an API Get Request If Soldier Is Alive, Expected 200:', () => {
         request(app).get('/health').expect("I am alive!");
     });
 });
+
+describe('Insert Soldier into DB #important: you need to modify the ID of inserted soldier in the test file in order to succeed this test', () => {
```

Comment:
describes are used for groups of tests, it looks as though you are placing each test in a describe. Cluster them in a different manner